### ISSUE TITLE
Unclear error when trying to boostrap without prior init

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#3909]
This is a followup to #3883. After upgrading to ae2807bd25ba36278f153860bb629b5e8a9496b0 and starting with (as per `README`):  `--gossip=NODE1:26257,NODE2,26257,...` I get:

```
I0117 17:11:26.081857 22348 gossip/gossip.go:911  starting client to 54.164.25.39:26257
I0117 17:11:26.081937 22348 storage/engine/rocksdb.go:106  opening rocksdb instance at "data"
W0117 17:11:26.083304 22348 rpc/client.go:320  rpc/codec/tls.go:62: unexpected HTTP response: 404 Not Found
I0117 17:11:26.164753 22348 server/node.go:280  store store=0:0 ([ssd]=data) not bootstrapped
I0117 17:11:26.164793 22348 gossip/gossip.go:267  read 0 gossip host(s) for bootstrapping from persistent storage
I0117 17:11:26.164808 22348 server/node.go:389  connecting to gossip network to verify cluster ID...
I0117 17:11:27.082166 22348 gossip/gossip.go:911  starting client to 54.152.118.20:26257
W0117 17:11:27.083761 22348 rpc/client.go:320  rpc/codec/tls.go:62: unexpected HTTP response: 404 Not Found
W0117 17:11:27.118872 22348 rpc/client.go:320  rpc/codec/tls.go:62: unexpected HTTP response: 404 Not Found
W0117 17:11:28.070934 22348 rpc/client.go:320  rpc/codec/tls.go:62: unexpected HTTP response: 404 Not Found
```

I also tried with `--gossip=tcp=NODE:26257` and with `http` but to no avail.

**Comment by knz** [comment]
Ok found the cause of the error: none of the nodes had been initialized, so there was no gossiping information to exchange in the first place.

Perhaps the message should be changed to: "no data available: either this peer has not joined the cluster yet or none of the peers was initialized"

@spencerkimball what do you think?

**Comment by vivekmenezes** [comment]
Looks like this should be a "uninitialized node failure". What exactly did
you forget to configure? Please do create an issue for it. Thanks

**Comment by spencerkimball** [comment]
Need to track down why we're getting the 404 error. I'll take this issue.

**Comment by knz** [comment]
@vivekmenezes  I was attempting to run start wihout running init first.

**Comment by vivekmenezes** [comment]
Thanks for reporting it!

**Comment by knz** [comment]
Fixed by #4027.
